spark-java/src/main/java/com/baeldung/sparkjava/UserServiceMapImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.sparkjava;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+
+public class UserServiceMapImpl  implements UserService{
+    private HashMap<String, User> userMap;
+    
+    public UserServiceMapImpl() {
+        userMap = new HashMap<>();
+    }
+
+    public void addUser (User user) {
+        userMap.put(user.getId(), user);
+    }
+    
+    public Collection<User> getUsers () {
+        return  userMap.values();
+    }
+    
+    public User getUser (String id) {
+        return userMap.get(id);
+    }
+    
+    public User editUser (String id, HashMap userArg) throws Exception{
```

Comment:
This implementation becomes trivial if you change the method to accept a User object instead of an id and Map.